pandas/core/reshape/reshape.py

```diff
@@ -1072,6 +1072,234 @@ def get_empty_frame(data) -> DataFrame:
         return DataFrame(dummy_mat, index=index, columns=dummy_cols)
 
 
+def from_dummies(
+    data: DataFrame,
+    to_series: bool = False,
+    prefix_sep: str | list[str] | dict[str, str] = "_",
+    columns: None | Index | list[str] = None,
+    dropped_first: None | str | list[str] | dict[str, str] = None,
+) -> Series | DataFrame:
+    """
+    Create a categorical `Series` or `DataFrame` from a `DataFrame` of dummy
+    variables.
+
+    Inverts the operation performed by 'get_dummies'.
+
+    Parameters
+    ----------
+    data : `DataFrame`
+        Data which contains dummy-coded variables.
+    to_series : bool, default False
+        Converts the input data to a categorical `Series`, converts the input data
+        to a categorical `DataFrame` if False.
+    prefix_sep : str, list of str, or dict of str, default '_'
+        Separator/deliminator used in the column names of the dummy categories.
+        Pass a list if multiple prefix separators are used in the columns names.
+        Will separate the prefix based on the first encountered separator following
+        the order of the list. Alternatively, pass a dictionary to map prefix
+        separators to prefixes if multiple and/or mixed separators are used in the
+        column names.
+    columns : None, Index, or list of str, default 'None'
+        The columns which to convert from dummy-encoding and return as categorical
+        `DataFrame`.
+        If `columns` is None then all dummy columns are converted and appended
+        to the non-dummy columns.
+    dropped_fist : None, str, list of str, or dict of str, default None
+        The implied value the dummy takes when all values are zero.
+        Can be a a single value for all variables, a list with a number of values
+        equal to the dummy variables, or a dict directly mapping the dropped value
+        to a prefix of a variable.
+
+    Returns
+    -------
+    `Series` or `DataFrame`
+        Categorical data decoded from the dummy input-data.
+
+    See Also
+    --------
+    get_dummies : Convert `Series` or `DataFrame` to dummy codes.
+
+    Examples
+    --------
+    >>> d = pd.DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0],
+    ...                   "c": [0, 0, 1, 0]})
+
+    >>> pd.from_dummies(s, to_series=True)
+    0  a
+    1  b
+    2  c
+    3  a
+
+    >>> d = pd.DataFrame({"C": [1, 2, 3], "col1_a": [1, 0, 1],
+    ...                   "col1_b": [0, 1, 0], "col2_a": [0, 1, 0],
+    ...                   "col2_b": [1, 0, 0], "col2_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(d)
```

Comment:
this is way too magical on the naming, e.g how is it dropping the _

Reply:
True, I will change it to always require a separator input if a separation is requred/wanted (This should be solved togehter with me removing `to_series`)